### ISSUE TITLE
Need help with RoyalMail Mailmark Barcode Generation

### ISSUE THREAD

**Issue by sanketbajoria** [woo-j/OkapiBarcode#72]
This is not a issue. 
I need to generate royalmail mailmark barcode. As far as i understood, We already have DataMatrix ISO/IEC 16022:2006. And, Mailmark barcode is also the same. Only requirement i can see, it enforces C40 encoding. And, customer data part can use alternative encoding. 

In current DataMatrix class, i am unable to enforced encoding. It get determined automatically. 

Can you please help me or point me in right direction, to start with

**Comment by gredler** [comment]
It's possible that the encoding selection algorithm will automatically select C40 for you -- I'd check that first. If it's not selecting C40, I'm not sure of the best approach. Per the spec, the C40 limitation does not apply to customer content, so you technically don't need to force C40 for the entire string. The lookAheadTest(...) method is the one deciding on any encoding changes, and generateCodewords( ) is where the modes are defaulted to ASCII to start.

**Comment by sanketbajoria** [comment]
@gredler 
Yes, customer info need not to be selecting C40. 

To start with, what i tried.
I have copied DataMatrix class and renamed it to MailMark.

Then,  I have modified lookAheadTest method to always return C40. And started generateCodewords default to C40. I have also kept customer content simple one, so that it can be encoded in C40. But, somehow, still barcode generated is not same one as https://barcode.tec-it.com/en/RoyalMail2D

Maybe, i have to do something else too.
